### PR TITLE
fix(kit): `InputNumber` ignores native attribute `maxlength`

### LINKED ISSUE #5058: 🐞 - `InputNumber` ignores native attribute `maxlength`
### Playground Link

https://stackblitz.com/edit/tui-input-number-maxlength-bug

### Description

```html
<tui-input-number>
    <input tuiTextfield maxlength="7" />
</tui-input-number>
```

Why?
https://github.com/Tinkoff/taiga-ui/blob/fe99d1377203508e01834f6acf3785885ecd20c6/projects/kit/components/input-number/input-number.directive.ts#L29

### Angular version

any

### Taiga UI version

3.39.0

### Which browsers have you used?

- [X] Chrome
- [ ] Firefox
- [ ] Safari
- [ ] Edge

### Which operating systems have you used?

- [X] macOS
- [ ] Windows
- [ ] Linux
- [ ] iOS
- [ ] Android

### PR DESCRIPTION
**Pull request by splincode**
## PR Type

What kind of change does this PR introduce?

- [x] Bugfix
- [ ] Feature
- [ ] Refactoring
- [ ] Build or CI related changes
- [ ] Tests related changes
- [ ] Documentation content changes

## What is the current behavior?

Closes #5058

### CODE REVIEW

File: projects/demo/src/modules/components/dialog/examples/9/index.html [hunk 1]
```diff
@@ -1,11 +1,14 @@
...
     <tui-input-number
+        tuiHintContent="The amount must be less than 1000 ₽"
+        tuiHintDirection="top"
         tuiTextfieldPostfix="₽"
         tuiTextfieldSize="l"
         [formControl]="amountControl"
         [tuiTextfieldCleaner]="true"
         [tuiTextfieldLabelOutside]="true"
     >
+        0 ... 999 ₽
```
Comment (vladimirpotekhin) [inline projects/demo/src/modules/components/dialog/examples/9/index.html:11]: do we need to add `maxlength` to this example?

Comment (splincode) [reply projects/demo/src/modules/components/dialog/examples/9/index.html:11]: But where?

Comment (nsbarsukov) [reply projects/demo/src/modules/components/dialog/examples/9/index.html:11]:
```html
<tui-input-number
        tuiHintContent="The amount must be less than 1000 ₽"
        tuiHintDirection="top"
        tuiTextfieldPostfix="₽"
        tuiTextfieldSize="l"
        [formControl]="amountControl"
        [tuiTextfieldCleaner]="true"
        [tuiTextfieldLabelOutside]="true"
    >
        0 ... 999 ₽
        <input
            maxlength="3"  <------------ here
            inputmode="numeric"
            tuiTextfield
        />
    </tui-input-number>
```

Comment (splincode) [reply projects/demo/src/modules/components/dialog/examples/9/index.html:11]: It doesn't work with maskito

Comment (nsbarsukov) [reply projects/demo/src/modules/components/dialog/examples/9/index.html:11]: `maxlength="5"`, my mistake)

**Review by nsbarsukov — APPROVED** [review]

**Review by waterplea — APPROVED** [review]

File: projects/kit/components/input-number/input-number.directive.ts [hunk 2]
```diff
@@ -26,7 +26,10 @@ export class TuiInputNumberDirective
             return;
         }
 
-        this.host.nativeFocusableElement.maxLength = this.host.calculatedMaxLength;
         this.host.nativeFocusableElement.inputMode = this.host.inputMode;
+        this.host.nativeFocusableElement.maxLength =
+            this.host.nativeFocusableElement.maxLength > 0
```
[the remark this answers is not in the document]
Comment (splincode) [reply projects/kit/components/input-number/input-number.directive.ts:31]: Yes, but default value as defined by W3C is  524288. However, I checked in some browsers and Safari/iOS, Chrome, Firefox browsers give -1

**Pull request merged**